src/System.Diagnostics.DiagnosticSource/ref/System.Diagnostics.DiagnosticSourceActivity.cs

```diff
@@ -9,18 +9,34 @@ namespace System.Diagnostics {
   public partial class Activity {
     public Activity(string operationName) {}      
     public string OperationName { get { throw null; } }
+
+    [System.ObsoleteAttribute]
     public string Id {get { throw null; } private set {} }
+
     public DateTime StartTimeUtc {get { throw null; } private set {} }
     public Activity Parent {get { throw null; } private set {} }
-    public string ParentId {get { throw null; } private set {} }
-    public string RootId {get { throw null; } private set {} }    
+  
+    public string ParentId {get { throw null; } set {} } // TraceId, public setter for binary (e.g. AMQP) or non-w3c custom propagation (e.g.Zipkin)
+
+    public string RootId {get { throw null; } set {} } // TraceId, public setter for binary (e.g. AMQP) or non-w3c custom propagation (e.g.Zipkin)
+
+    public string SpanId { get { throw null; } private set { } }
+
+    public byte TraceFlags {get { throw null; } set {} }
+
+    public string Tracestate {get { throw null; } set {} }
+
+    public string W3CId { get { throw null; } set { } }    // 00-traceid-spanid-01
     public TimeSpan Duration {get { throw null; } private set {} }    
     public System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, string>> Tags { get { throw null; } }    
     public System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, string>> Baggage { get { throw null; } }
     public string GetBaggageItem(string key) {throw null;}
     public Activity AddTag(string key, string value) {throw null;}
     public Activity AddBaggage(string key, string value) {throw null;}
+
+    [System.ObsoleteAttribute]
```

Comment:
I think we should not mark this as obsolete, but maybe I misunderstand your scenario.